### PR TITLE
CARTO provider for cloud storage

### PR DESCRIPTION
**Pull request by neokore**
This feature allows loading/saving private and public maps using CARTO.

Related to https://github.com/keplergl/kepler.gl/issues/804

### CODE REVIEW

File: examples/demo-app/src/cloud-providers/carto/carto-icon.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  static defaultProps = {
+    height: '100%',
+    viewBox: '0 0 92 36',
+    predefinedClassName: 'data-ex-icons-carto'
+  };
+
+  render() {
+    return (
+      <Icons.IconWrapper {...this.props} viewBox={'0 0 92 36'}>
```
Comment (heshan0131) [inline examples/demo-app/src/cloud-providers/carto/carto-icon.js]: `IconWrapper` should pass viewBox from props, ` viewBox={'0 0 92 36'}` shouldn't be needed here

File: examples/demo-app/src/cloud-providers/carto/carto-provider.js [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+      },
+      {
+        serverUrlTemplate: `https://{user}.${DOMAIN}/`,
+        namespace: NAMESPACE
+      }
+    );
+
+    this._carto.setClientID(clientId);
+
+    if (this.getAccessToken()) {
```
Comment (heshan0131) [inline examples/demo-app/src/cloud-providers/carto/carto-provider.js]: should it be `!this.getAccessToken()`?

File: examples/demo-app/src/cloud-providers/carto/carto-provider.js [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+    } catch (error) {
+      this._manageErrors(error);
+    }
+  }
+
+  isEnabled() {
+    return this.clientId !== null;
+  }
+
+  isConnected() {
```
Comment (heshan0131) [inline examples/demo-app/src/cloud-providers/carto/carto-provider.js]: i don't think `isConnected` is used anymore.  in the UI we always call `const isConnected = Boolean(cloudProvider.getAccessToken()`

File: examples/demo-app/src/cloud-providers/carto/carto-provider.js [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+    } catch (error) {
+      this._manageErrors(error);
+    }
+  }
+
+  /**
+   * The CARTO cloud provider sets by the default the privacity to public
+   * @param {*} metadata
+   */
+  shareFile(metadata) {
```
Comment (heshan0131) [inline examples/demo-app/src/cloud-providers/carto/carto-provider.js]: this method is private to dropbox, it is not called by UI

File: examples/demo-app/src/cloud-providers/carto/carto-provider.js [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+
+      formattedVis = formattedVis.sort((a, b) => b.lastModification - a.lastModification);
+
+      return formattedVis;
+    } catch (error) {
+      this._manageErrors(error);
+    }
+  }
+
+  getMapPermalink(mapLink, fullURL = true) {
```
Comment (heshan0131) [inline examples/demo-app/src/cloud-providers/carto/carto-provider.js]: no longer called by UI, can make it private

File: examples/demo-app/src/cloud-providers/carto/carto-provider.js [hunk 6]
```diff
@@ -0,0 +1,424 @@
...
+    return fullURL
+      ? `${window.location.protocol}//${window.location.host}/${mapLink}`
+      : `/${mapLink}`;
+  }
+
+  getMapPermalinkFromParams({mapId, owner, privateMap}, fullURL = true) {
```
Comment (heshan0131) [inline examples/demo-app/src/cloud-providers/carto/carto-provider.js]: no longer called by UI, can make it private

File: examples/demo-app/src/cloud-providers/carto/carto-icon.js (hunk 1 above)
Comment (neokore) [reply examples/demo-app/src/cloud-providers/carto/carto-icon.js]: Fixed!

File: examples/demo-app/src/cloud-providers/carto/carto-provider.js (hunk 3 above)
Comment (neokore) [reply examples/demo-app/src/cloud-providers/carto/carto-provider.js]: 👍

File: examples/demo-app/src/cloud-providers/carto/carto-provider.js (hunk 4 above)
Comment (neokore) [reply examples/demo-app/src/cloud-providers/carto/carto-provider.js]: 👍

File: examples/demo-app/src/cloud-providers/carto/carto-provider.js (hunk 6 above)
Comment (neokore) [reply examples/demo-app/src/cloud-providers/carto/carto-provider.js]: 👍

File: examples/demo-app/src/cloud-providers/carto/carto-provider.js (hunk 5 above)
Comment (neokore) [reply examples/demo-app/src/cloud-providers/carto/carto-provider.js]: 👍 Not needed anymore, deleted

File: examples/demo-app/src/cloud-providers/carto/carto-provider.js (hunk 2 above)
Comment (neokore) [reply examples/demo-app/src/cloud-providers/carto/carto-provider.js]: 👍 This is not needed anymore, I deleted it

**Pull request merged**